Create Basic Components for Quick Create Section

knittingarch opened the pull request:
The components created here are placeholders for more robust design and
state.  Currently mocking up the components I will need for the MVP and
then will add state and interactivity as well as design.

The Quick Create section will be the top level way to create a task, but
users will also be able to click "more" if they want to add a set date or
frequency.  The defaults will be the current date and non-repeating.

<img width="183" alt="Screen Shot 2019-10-04 at 5 50 49 PM" src="https://user-images.githubusercontent.com/7423975/66243325-35b3b080-e6d2-11e9-970f-c0ad680a5f6c.png">

knittingarch committed:
Revert "Create Basic Components for Quick Create Section"

This reverts commit f7fb1e48518cf8b2768484524dedcdad40b0d64b.

knittingarch committed:
Create Basic Components for Quick Create Section

The components created here are placeholders for more robust design and
state.  Currently mocking up the components I will need for the MVP and
then will add state and interactivity as well as design.

The Quick Create section will be the top level way to create a task, but
users will also be able to click "more" if they want to add a set date or
frequency.  The defaults will be the current date and non-repeating.

knittingarch commented:
So I switched to master to check something and forgot to change back to this branch.  I messed up the fix so the files are listed under the second commit :/ Sigh.

sidraval commented on src/AddTaskButton.js:
```diff
@@ -0,0 +1,9 @@
+  )
+}
+
+export default AddTaskButton
```
Above on line 3, you can write `export default function AddTaskButton() { ....` export/define in the same declaration

sidraval commented on src/LaunchModalButton.js:
```diff
@@ -0,0 +1,11 @@
+import React from "react"
+
+class LaunchModalButton extends React.Component {
```
This can be a functional component, because it's stateless

knittingarch replied:
It's stateless now, but will have state shortly.  Is it better practice to write functional components and only convert once state is introduced?

sidraval reviewed (approved):
Couple small comments, but looking good overall.

The comment about exporting/defining the functional component in the same declaration can be applied across most of your files.

One tool I see used in almost every JavaScript project these days is [prettier](https://github.com/prettier/prettier) for automated code formatting. I personally love it, because it takes all the formatting decisions out of the code.

knittingarch committed:
Move `export default` And Add Prettier

This commit moves the explicit export at the end of each component to the function declaration.

It also adds `prettier` to handle auto-formatting.
